src/cmd/flash.js

```diff
@@ -24,16 +27,42 @@ module.exports = class FlashCommand {
 
 		let result;
 		if (usb){
-			result = this.flashDfu({ binary, factory, force });
+			if (files.length > 0) {
+				// If both a device and a binary were provided, the binary will come in as the
+				// first element in the files array
+				binary = files[0];
```

Comment:
this is ambiguous at best and likely inaccurate. i've been adding tests like [this](https://github.com/particle-iot/particle-cli/pull/565/commits/33595ec6fc136038318562fa7d6e805dab817e47) to help clarify similar behavior.